readthedocs/api/v3/views.py

```diff
@@ -300,8 +283,11 @@ def get_queryset(self):
         # ``super().get_queryset`` produces the filter by ``NestedViewSetMixin``
         queryset = super().get_queryset()
 
-        # the user can only see profiles from people they share a project with
-        queryset = queryset.filter(
-            projects__in=Project.objects.api(user=self.request.user),
-        ).distinct()
-        return queryset
+        if self.detail:
+            return queryset
+
+        # give access to the user if it's maintainer of the project
+        if self.request.user in self.model.objects.filter(**self.get_parents_query_dict()):
```

Comment:
This should be either a queryset method or a mixin. I think as a rule we should have _no_ custom querysets unless we have a really good reason.

Reply:
I refactored all the auth/permission checks into a Mixin and a Permission class. Please, let me know what you think about the new updates.